pkg/daemon/daemon.go

```diff
@@ -201,19 +201,25 @@ func New(
 		err        error
 	)
 
-	os := OperatingSystem{}
+	hostos := OperatingSystem{}
 	if !mock {
-		os, err = GetHostRunningOS()
+		hostos, err = GetHostRunningOS()
 		if err != nil {
 			HostOS.WithLabelValues("unsupported", "").Set(1)
 			return nil, errors.Wrapf(err, "checking operating system")
 		}
 	}
 
 	// Only pull the osImageURL from OSTree when we are on RHCOS or FCOS
-	if os.IsCoreOSVariant() {
+	if hostos.IsCoreOSVariant() {
 		osImageURL, osVersion, err = nodeUpdaterClient.GetBootedOSImageURL()
 		if err != nil {
+			// If this fails for some reason, let's dump the unit status
+			// into our logs to aid future debugging.
+			cmd := exec.Command("systemctl", "status", "rpm-ostreed")
```

Comment:
Does printing status will capture enough error logs or should we print here rpm-ostreed journal log from current boot?

Reply:
Status will capture the last few lines of logs at the point in time this happens, which I think will be enough.  

https://github.com/openshift/must-gather/pull/244 will help for the general case.